test/unit/math/mix/mat/seq_reader.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef TEST_UNIT_MATH_MIX_MAT_SEQ_READER_HPP
+#define TEST_UNIT_MATH_MIX_MAT_SEQ_READER_HPP
+
+#include <stan/math/mix/mat.hpp>
+#include <stdexcept>
+
+namespace stan {
+  namespace math {
+    namespace test {
+
+      template <typename T>
+      struct seq_reader {
+        typedef Eigen::Matrix<T, -1, -1> matrix_t;
+        typedef Eigen::Matrix<T, -1, 1> vector_t;
+        typedef Eigen::Matrix<T, 1, -1> row_vector_t;
+
+        vector_t data_;
+        int pos_;
+
+        seq_reader(const vector_t& data) : data_(data), pos_(0) { }
+
+        T next() {
+          if (pos_ >= data_.size())
+            throw std::logic_error("attempt to read past end of reader");
+          return data_[pos_++];
+        }
+
+        /**
+         * Read an object with size and container type determined by
+         * the specified argument and return scalar type T determined
+         * by the class instantiation.
+         */
+        T read(double x) {
+          return next();
+        }
+
+        template <int R, int C>
+        matrix_t read(const Eigen::Matrix<double, R, C>& x) {
```

Comment:
I realize these vectorized versions of `read` aren't used anywhere yet but it might be worth noting in the docstring maybe here and maybe in things that use `binary_binder` that the `x` being read into must already have its size specified before the call.